lib/dal/src/component.rs

```diff
@@ -1732,24 +1765,35 @@ impl Component {
         let vet = Self::input_socket_attribute_values_for_component_id(ctx, component_id).await?;
         for (_, input_socket_match) in vet {
             if let Some(output_socket) =
-                Component::find_inferred_connection_to_input_socket(ctx, input_socket_match).await?
+                Component::find_potential_inferred_connection_to_input_socket(
+                    ctx,
+                    input_socket_match,
+                )
+                .await?
             {
                 results.entry(input_socket_match).or_insert(output_socket);
             }
         }
+        debug!(
+            "Map of inferred input to output connections for component {:?}: {:?}",
+            component_id, results
+        );
         Ok(results)
     }
     /// For a given [`InputSocketMatch`], find the single inferred [`OutputSocketMatch`] that is driving it
     /// if it exists. This walks up or down the component lineage tree depending on the [`ComponentType`]
     /// and finds the closest matching [`OutputSocket`]
-    #[instrument(level = "debug", skip_all)]
-    pub async fn find_inferred_connection_to_input_socket(
+    ///
+    /// Note: this does not check for whether data should actually flow between components
+    #[instrument(level = "debug", skip(ctx))]
+    pub async fn find_potential_inferred_connection_to_input_socket(
```

Comment:
I renamed some of these functions as now, whether there is an inferred connection is a separate concern from whether the data should flow through.